File: src/main/scala/warmUp/Java/DiagonalDifference.java
/**
 * @author Ram Saran Vuppuluri
 * Problem statement
 * @see https://www.hackerrank.com/challenges/diagonal-difference/problem
 */
package warmUp.Java;

import java.util.Scanner;

public class DiagonalDifference {
    public static void main(String... args) {
        Scanner scanner = new Scanner(System.in);
        int matrixDimension = scanner.nextInt();
        int sumOfDimension1 = 0;
        int sumOfDimension2 = 0;

        for (int i = 1; i <= matrixDimension; i++) {
            for (int j = 1; j <= matrixDimension; j++) {
                int nextInt = scanner.nextInt();
                if (i == j) {
                    sumOfDimension1 += nextInt;
                }
                if ((i + j) == (matrixDimension + 1)) {
                    sumOfDimension2 += nextInt;
                }
            }
        }

        if ((sumOfDimension1 - sumOfDimension2) >= 0) {
            System.out.println(sumOfDimension1 - sumOfDimension2);
        } else {
            System.out.println((sumOfDimension1 - sumOfDimension2) * -1);
        }
    }
}

File: src/main/scala/warmUp/Scala/DiagonalDifference.scala
/**
  * @author Ram Saran Vuppuluri
  *         Problem statement
  * @see https://www.hackerrank.com/challenges/diagonal-difference/problem
  */
package warmUp.Scala

import java.util.Scanner

object DiagonalDifference extends App {
  val scanner = new Scanner(System.in)
  val matrixDimension = scanner.nextInt()
  var diagnal1Sum = 0
  var diagnal2Sum = 0
  for (
    i <- 1 to matrixDimension;
    j <- 1 to matrixDimension
  ) {
    var nextInt: Int = scanner.nextInt()
    if (i == j) {
      diagnal1Sum += nextInt
    }
    if ((i + j) == (matrixDimension + 1)) {
      diagnal2Sum += nextInt
    }
  }

  if ((diagnal1Sum - diagnal2Sum) >= 0) {
    println(diagnal1Sum - diagnal2Sum)
  } else {
    println(((diagnal1Sum - diagnal2Sum) * (-1)))
  }
}